client/lib/mixins/url-search/component.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import assign from 'lodash/assign';
+
+/**
+ * Internal dependencies
+ */
+import { doSearch, getSearchOpen } from './index';
+
+export default function( Component ) {
+	const componentName = Component.displayName || Component.name || '';
+
+	return React.createClass( {
+		displayName: 'Searchable' + componentName,
+
+		getInitialState() {
+			return {
+				searchOpen: false
+			};
+		},
+
+		componentWillReceiveProps( nextProps ) {
+			if ( ! nextProps.search ) {
+				this.setState( {
+					searchOpen: false
+				} );
+			}
+		},
```

Comment:
do we only want to change the state if the search prop doesn't exist? what about turning it back on?

either way, here is a way we could shorten this up

```js
componentWillReceiveProps: ( { search } ) => ! search && this.setState( { searchOpen: false } );
```